packages/app-store/_components/eventTypeAppCardInterface.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { render, screen } from "@testing-library/react";
+import type { CredentialOwner } from "types";
+import { vi } from "vitest";
+
+import EventTypeAppContext from "@calcom/app-store/EventTypeAppContext";
+import type { RouterOutputs } from "@calcom/trpc";
+
+import { DynamicComponent } from "./DynamicComponent";
+import { EventTypeAppCard } from "./EventTypeAppCardInterface";
+
+vi.mock("./DynamicComponent", async () => {
+  const actual = (await vi.importActual("./DynamicComponent")) as object;
+  return {
+    ...actual,
+    DynamicComponent: vi.fn(() => <div>MockedDynamicComponent</div>),
+  };
+});
+
+afterEach(() => {
+  vi.clearAllMocks();
+});
+
+const getAppDataMock = vi.fn();
+const setAppDataMock = vi.fn();
+const mockProps = {
+  app: {
+    name: "TestApp",
+    slug: "testapp",
+    credentialOwner: {},
+  } as RouterOutputs["viewer"]["integrations"]["items"][number] & { credentialOwner?: CredentialOwner },
+  eventType: {},
+  getAppData: getAppDataMock,
+  setAppData: setAppDataMock,
+  LockedIcon: <div>MockedIcon</div>,
+  disabled: false,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+} as any;
+
+describe("Tests for EventTypeAppCard component", () => {
+  test("Should render DynamicComponent with correct slug", () => {
+    render(<EventTypeAppCard {...mockProps} />);
+
+    expect(DynamicComponent).toHaveBeenCalledWith(
```

Comment:
I feel we should test that `EventTypeAppCardInterface` of the appropriate app has been rendered here with correct props. It would test `DynamicComponent` as well in that way. But I think that's fine because that's what the purpose of the `EventTypeAppCardInterface` component is, it doesn't render anything of it's own in UI, it just delegates to the respective app's EventTypeAppCardInterface

Reply:
Also, we should test stripe app as one of the apps as that has a special handling in the code.
<img width="893" alt="Screenshot 2023-09-15 at 3 40 03 PM" src="https://github.com/calcom/cal.com/assets/1780212/80889be8-6346-4cd5-b236-f9015a964657">

Reply:
Alright, we will implement it